src/Mutant.ts

```diff
@@ -89,29 +86,62 @@ export default class Mutant {
   };
 
   /**
+   * @param mutation - The mutation which was applied to this Mutant.
    * @param filename - The name of the file which was mutated, including the path.
    * @param originalCode - The original content of the file which has not been mutated.
-   * @param mutation - The mutation which was applied to this Mutant.
-   * @param ast - The abstract syntax tree of the file.
-   * @param node - The part of the ast which has been mutated.
-   * @param columnNumber - The column which has been mutated.
+   * @param substitude - The mutated code which will replace a part of the originalCode.
+   * @param mutatedLocation - The part of the originalCode which has been mutated.
    */
-  constructor(private _filename: string, originalCode: string, private _mutation: BaseMutation, ast: ESTree.Program, node: ESTree.Node, private _columnNumber: number) {    
-    this._lineNumber = node.loc.start.line;
-    this._mutatedCode = this.parserUtils.generate(ast, originalCode);
+  constructor(private _mutation: BaseMutation, private _filename: string, private _originalCode: string, substitude: string, private mutatedLocation: ESTree.SourceLocation) {
     this.status = MutantStatus.UNTESTED;
-    this._mutatedLine = _.trim(this.mutatedCode.split('\n')[this._lineNumber - 1]);
-    this._originalLine = _.trim(originalCode.split('\n')[this._lineNumber - 1]);
+
+    this.insertSubstitude(substitude);
+
     this.save();
   }
 
   /**
+   * Inserts the substitude into the mutatedCode based on the mutatedLocation.
+   * This also alters the originalLine and mutatedLine.
+   * @param substitude - The mutated code which will replace a part of the originalCode
+   */
+  private insertSubstitude(substitude: string) {
+    let linesOfCode = this._originalCode.split('\n');
+
+    if (this.mutatedLocation.start.line === this.mutatedLocation.end.line) {
```

Comment:
I think this if can be removed. The multi-line replacing logic will also work for single line mutations